drivers/include/sx1276.h

```diff
@@ -0,0 +1,456 @@
+    /**< SX1276 DIO interrupt lines events handler stack */
+    uint8_t dio_polling_thread_stack[SX1276_EVENT_HANDLER_STACK_SIZE];
+
+    /* Timers */
+    xtimer_t tx_timeout_timer;              /**< TX operation timeout timer */
+    xtimer_t rx_timeout_timer;              /**< RX operation timeout timer */
+} sx1276_internal_t;
+
+/**
+ * SX1276 hardware and global parameters.
+ */
+typedef struct sx1276_s {
+    spi_t spi;                                                          /**< SPI */
+    gpio_t nss_pin;                                                     /**< SPI NSS pin */
+
+    gpio_t reset_pin;                                                   /**< Reset pin */
+    gpio_t dio0_pin;                                                    /**< Interrupt line DIO0 */
+    gpio_t dio1_pin;                                                    /**< Interrupt line DIO1 */
+    gpio_t dio2_pin;                                                    /**< Interrupt line DIO2 */
+    gpio_t dio3_pin;                                                    /**< Interrupt line DIO3 */
+    gpio_t dio4_pin;                                                    /**< Interrupt line DIO4 (not used) */
+    gpio_t dio5_pin;                                                    /**< Interrupt line DIO5 (not used) */
+
+    sx1276_settings_t settings;                                         /**< Transceiver settings */
+
+    void (*sx1276_event_cb)(void *dev, sx1276_event_type_t event_type); /**< Event callback */
+    void *callback_arg;                                                 /**< User-defined callback argument */
+
+    sx1276_internal_t _internal;                                        /**< Internal sx1276 data used within the driver */
+} sx1276_t;
+
+/**
+ * SX1276 initialization result.
+ */
+typedef enum {
+    SX1276_INIT_OK = 0,         /**< Initialization was successful */
+    SX1276_ERR_SPI,             /**< Failed to initialize SPI bus or CS line */
+    SX1276_ERR_TEST_FAILED,     /**< SX1276 testing failed during initialization (check chip) */
+    SX1276_ERR_THREAD           /**< Unable to create DIO handling thread (check amount of free memory) */
+} sx1276_init_result_t;
+
+/**
+ * Hardware IO IRQ callback function definition.
+ */
+typedef void (sx1276_dio_irq_handler_t)(sx1276_t *dev);
+
+/**
+ * SX1276 definitions.
+ */
+#define SX1276_XTAL_FREQ       32000000 /**< 32MHz */
+#define SX1276_FREQ_STEP       61.03515625
+
+/**
+ * Public functions prototypes.
+ */
+
+/**
+ * @brief Tests the transceiver.
+ *
+ * @param	[IN]	dev	The sx1276 device structure pointer
+ * @return true if test passed, false otherwise
+ */
+bool sx1276_test(sx1276_t *dev);
+
+/**
+ * @brief Initializes the transceiver.
+ *
+ * @param	[IN]	dev					The sx1276 device structure pointer
+ *
+ * @return result of initialization (see @sx1276_init_result_t struct)
+ */
+
+sx1276_init_result_t sx1276_init(sx1276_t *dev);
```

Comment:
Could you please use the same scheme for initialization as done for other network devices, e.g. `at86rf2xx` or `w5100`:
- use a setup function for initializing the device descriptor
```c
void xs1276_setup(sx1276_t *dev, const sx1276_params_t *params);
```
- and a stand-alone init function to be called later, once the context ist setup
```c
int sx1276_init(sx1276_t *dev);
```
?!